core/cmd/local_client.go

```diff
@@ -214,6 +226,42 @@ func logConfigVariables(store *strpkg.Store) error {
 	return nil
 }
 
+func setupFundingKey(ctx context.Context, str *strpkg.Store, pwd string) (*models.Key, *big.Int, error) {
+	key := models.Key{}
+	err := str.DB.Where("is_funding = TRUE").First(&key).Error
+	if err != nil && err.Error() != "record not found" {
+		return nil, nil, err
+	}
+	if err == nil && key.ID != 0 {
+		// TODO How to make sure the EthClient is connected?
+		balance, ethErr := str.EthClient.BalanceAt(ctx, gethCommon.HexToAddress(string(key.Address)), nil)
+		return &key, balance, ethErr
+	}
+	// Key record not found so create one.
+	ethAccount, err := str.KeyStore.NewAccount(pwd)
+	if err != nil {
+		return nil, nil, err
+	}
+	exportedJSON, err := str.KeyStore.Export(ethAccount, pwd, pwd)
+	if err != nil {
+		return nil, nil, err
+	}
+	var firstNonce int64 = 1
+	key = models.Key{
+		Address:   models.EIP55Address(ethAccount.Address.Hex()),
+		IsFunding: true,
+		JSON: models.JSON{
+			Result: gjson.ParseBytes(exportedJSON),
+		},
+		NextNonce: &firstNonce,
+	}
+	if err = str.CreateKeyIfNotExists(key); err != nil {
```

Comment:
Given that we just generated it, why would it already exist at this point?

Reply:
Tbh, only because this method already existed. 
I agree it's a bit confusing, but in terms of performance, this method perform an INSERT https://github.com/smartcontractkit/chainlink/blob/develop/core/store/orm/orm.go#L1270